pkg/nmstate/nmstateconfig.go

```diff
@@ -0,0 +1,162 @@
+package nmstate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	assistedv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Builder provides struct for the NMStateConfig object containing connection to
+// the cluster and the NMStateConfig definitions.
+type NmStateConfigBuilder struct {
+	// NMStateConfig definition. Used to create NMStateConfig object with minimum set of required elements.
+	Definition *assistedv1beta1.NMStateConfig
+	// Created NMStateConfig object on the cluster.
+	Object *assistedv1beta1.NMStateConfig
+	// API client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before NMStateConfig object is created.
+	errorMsg string
+}
+
+// NewBuilder creates a new instance of NMStateConfig Builder.
+func NewNmStateConfigBuilder(apiClient *clients.Settings, name, namespace string) *NmStateConfigBuilder {
+	glog.V(100).Infof("Initializing new NMStateConfig structure with the name: %s", name)
+
+	builder := NmStateConfigBuilder{
+		apiClient: apiClient,
+		Definition: &assistedv1beta1.NMStateConfig{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the NMStateConfig is empty")
+
+		builder.errorMsg = "NMStateConfig 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Exists checks whether the given NMStateConfig exists.
+func (builder *NmStateConfigBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if NMStateConfig %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	if err != nil {
```

Comment:
I don't think we need a separate conditional for a glog. Error checking the return value of this method should provide why the error occurred